### ISSUE TITLE
Show only relevant POIs on map when searching

### ISSUE THREAD

**Issue by arnaudvergnet** [organicmaps/organicmaps#2385]
Currently OM simply highlights results when showing search results on the map. The map then looks very cluttered when in a city. Other map applications and websites show only the results and some important landmarks such as public transport stops.

Here is a comparison between Om and Qwant maps (using OSM as well) as an example (taken [here](https://www.qwant.com/maps/places/?type=restaurant#map=17.20/43.6055020/1.4489054)):

| OM | Qwant Maps |
|-|-|
| ![Screenshot_20220412-083209_Organic_Maps](https://user-images.githubusercontent.com/80701113/162895513-d9ec1e27-07e4-4db9-bb7a-ac54a1100e8a.png) | ![Screenshot 2022-04-12 at 08-32-44 Qwant Maps](https://user-images.githubusercontent.com/80701113/162895589-44b16909-1065-4812-9de5-40a9a78c0e05.png) |

 Similar to #2355 and could be done easily when such feature is implemented.

**Comment by biodranik** [comment]
If I understood you correctly, you offer to "simplify" the map rendering style under the highlighted POIs, right?

**Comment by pastk** [comment]
Maybe instead of completely removing the icons they could be made half-transparent / greyed out and stripped of labels? This way the POIs won't get into the eyes as much but will be available for orienteering still.

Also the search results should show their text labels.

**Comment by arnaudvergnet** [comment]
Making other icons half transparent is a bad idea as the user would think they are disabled. Giving more importance to search results and showing them bigger in bold font would be a first step. This way if you zoom in enough you will see other icons but they should not appear on lower zoom levels if there are search results nearby as they do not have the rendering priority.

**Comment by pastk** [comment]
Right, I think its just a matter of making search results displace nearby POIs (just like the way nearby POIs are displacing each other based on priority).
At the moment there is no displacement happening as the search results are treated as an independent top layer and the visual mess is created by search results covering parts of POIs.

With the displacement mechanism only the POIs not intersecting with the search results will be visible and they will provide a context to orienteer around.

**Comment by oleg-rswll** [comment]
Related to #5675
